### ISSUE TITLE
Invalid type are not detected as error

### ISSUE THREAD

**Issue by LasneF** [daveshanley/vacuum#411]
given the API spec below 

the  type : jojo should have been spot as an error , but looks not spot 

version 
leveraging Vaccum online 

```
{
    "openapi": "3.1.0",
    "info": {
        "title": "this is the title",
        "description": "this is the description",
        "version": "1.0"
    },
    "servers": [
        {
            "url": "https://api.server.test/v1"
        }
    ],
    "paths": {
        "/test": {
            "get": {
                "operationId": "getTest",
                "responses": {
                    "200": {
                        "description": "testOK",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Pet"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "Pet": {
                "type": "jojo"
            }
        }
    }
}
```

**Comment by daveshanley** [comment]
strange, will investigate.

**Comment by daveshanley** [comment]
This has been fixed in `v0.8.0` with the introduction of a new rule `oas-schema-check`. It will perform a type check and perform basic validation on that type compared with other properties on the schema. 

Docs are here: https://quobix.com/vacuum/rules/schemas/oas-schema-check/
